src/OrchardCore.Modules/OrchardCore.Email/Workflows/Activities/EmailTask.cs

```diff
@@ -112,8 +153,8 @@ public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecuti
             // Email reply-to header https://tools.ietf.org/html/rfc4021#section-2.1.4
             ReplyTo = replyTo?.Trim(),
             Subject = subject?.Trim(),
-            Body = body?.Trim(),
-            IsHtmlBody = IsHtmlBody
+            HtmlBody = htmlBody?.Trim(),
```

Comment:
If you want to make the message format selectable, you have to evaluate this here as well. Otherwise, a user can type e.g. HTLM, and if they later switch to plain text, the HTML body will still be sent.

Reply:
Just wondering if we can make this change on the driver or in the task, the advantage of doing it here we could preserve both bodies even if one of them is selected